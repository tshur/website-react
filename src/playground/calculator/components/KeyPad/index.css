:local .row {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  grid-gap: 15px;
  margin-bottom: 15px;
}