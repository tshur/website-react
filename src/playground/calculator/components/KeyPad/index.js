import React from 'react';
import styles from './index.css';

import Key from '../Key';

const KeyPad = ({ onClick, operator, current }) => {
  let clear_type = current || operator ? 'C' : 'AC';
  return (
    <div>
      <div className={styles.row}>
        <Key symbol={clear_type} onClick={() => onClick(clear_type)} />
        <Key symbol='+/-' onClick={() => onClick('+/-')} />
        <Key symbol='%' onClick={() => onClick('%')} />
        <Key symbol='/' onClick={() => onClick('/')} operator={operator} />
      </div>

      <div className={styles.row}>
        <Key symbol='7' onClick={() => onClick('7')} />
        <Key symbol='8' onClick={() => onClick('8')} />
        <Key symbol='9' onClick={() => onClick('9')} />
        <Key symbol='*' onClick={() => onClick('*')} operator={operator} />
      </div>

      <div className={styles.row}>
        <Key symbol='4' onClick={() => onClick('4')} />
        <Key symbol='5' onClick={() => onClick('5')} />
        <Key symbol='6' onClick={() => onClick('6')} />
        <Key symbol='-' onClick={() => onClick('-')} operator={operator} />
      </div>

      <div className={styles.row}>
        <Key symbol='1' onClick={() => onClick('1')} />
        <Key symbol='2' onClick={() => onClick('2')} />
        <Key symbol='3' onClick={() => onClick('3')} />
        <Key symbol='+' onClick={() => onClick('+')} operator={operator} />
      </div>

      <div className={styles.row}>
        <Key symbol='0' onClick={() => onClick('0')} />
        <Key symbol='.' onClick={() => onClick('.')} />
        <Key symbol='=' onClick={() => onClick('=')} />
      </div>
    </div>
  );
}

export default KeyPad;
