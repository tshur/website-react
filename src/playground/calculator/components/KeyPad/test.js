import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import KeyPad from '.';

describe('KeyPad', () => {

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<KeyPad />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test('has a valid snapshot', () => {
    const wrapper = shallow(
      <KeyPad />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
