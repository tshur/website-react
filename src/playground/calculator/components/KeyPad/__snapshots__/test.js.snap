// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KeyPad has a valid snapshot 1`] = `
<div>
  <div>
    <Key
      onClick={[Function]}
      symbol="C"
    />
    <Key
      onClick={[Function]}
      symbol="+/-"
    />
    <Key
      onClick={[Function]}
      symbol="%"
    />
    <Key
      onClick={[Function]}
      symbol="/"
    />
  </div>
  <div>
    <Key
      onClick={[Function]}
      symbol="7"
    />
    <Key
      onClick={[Function]}
      symbol="8"
    />
    <Key
      onClick={[Function]}
      symbol="9"
    />
    <Key
      onClick={[Function]}
      symbol="*"
    />
  </div>
  <div>
    <Key
      onClick={[Function]}
      symbol="4"
    />
    <Key
      onClick={[Function]}
      symbol="5"
    />
    <Key
      onClick={[Function]}
      symbol="6"
    />
    <Key
      onClick={[Function]}
      symbol="-"
    />
  </div>
  <div>
    <Key
      onClick={[Function]}
      symbol="1"
    />
    <Key
      onClick={[Function]}
      symbol="2"
    />
    <Key
      onClick={[Function]}
      symbol="3"
    />
    <Key
      onClick={[Function]}
      symbol="+"
    />
  </div>
  <div>
    <Key
      onClick={[Function]}
      symbol="0"
    />
    <Key
      onClick={[Function]}
      symbol="."
    />
    <Key
      onClick={[Function]}
      symbol="="
    />
  </div>
</div>
`;
